### PR TITLE
cleanups as suggested by errorprone

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR applies errorprone as in below diff and applies the suggested changes for the first few projects. The permanent application of errorProne is removed within the scope of this PR as most projects will give positives.

```
commit 9f9a2b6b7e094537fea75f633d355f983d4c15d7
Author: Daan Hoogland <daan@onecht.net>
Date:   Thu Jan 12 19:32:35 2023 +0100

    errorprone

diff --git a/pom.xml b/pom.xml
index 204c74cc76..13c46d7521 100644
--- a/pom.xml
+++ b/pom.xml
@@ -817,6 +817,32 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${cs.compiler-plugin.version}</version>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
+                    <encoding>UTF-8</encoding>
+                    <compilerArgs>
+                        <arg>-XDcompilePolicy=simple</arg>
+                        <arg>-Xplugin:ErrorProne</arg>
+                    </compilerArgs>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>com.google.errorprone</groupId>
+                            <artifactId>error_prone_core</artifactId>
+                            <version>2.18.0</version>
+                        </path>
+                        <!-- Other annotation processors go here.
+
+                        If 'annotationProcessorPaths' is set, processors will no longer be
+                        discovered on the regular -classpath; see also 'Using Error Prone
+                        together with other annotation processors' below. -->
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
         </plugins>
         <pluginManagement>
             <plugins>
@@ -1074,15 +1100,26 @@
                     <configuration>
                         <source>${cs.jdk.version}</source>
                         <target>${cs.jdk.version}</target>
-                        <fork>true</fork>
-                        <meminitial>128m</meminitial>
-                        <maxmem>512m</maxmem>
+                        <encoding>UTF-8</encoding>
                         <compilerArgs>
                             <arg>-XDignore.symbol.file=true</arg>
                             <arg>--add-opens=java.base/java.lang=ALL-UNNAMED</arg>
                             <arg>--add-exports=java.base/sun.security.x509=ALL-UNNAMED</arg>
                             <arg>--add-exports=java.base/sun.security.provider=ALL-UNNAMED</arg>
+                            <arg>-XDcompilePolicy=simple</arg>
+                            <arg>-Xplugin:ErrorProne</arg>
+                            <arg>--illegal-access=warn</arg>
                         </compilerArgs>
+                        <annotationProcessorPaths>
+                            <path>
+                                <groupId>com.google.errorprone</groupId>
+                                <artifactId>error_prone_core</artifactId>
+                                <version>2.18.0</version>
+                            </path>
+                        </annotationProcessorPaths>
+                        <fork>true</fork>
+                        <meminitial>128m</meminitial>
+                        <maxmem>512m</maxmem>
                     </configuration>
                 </plugin>
                 <plugin>
```

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [x] CI/build
#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

Testing should be usual regression and monkey testing

### CODE REVIEW

**Comment by DaanHoogland** [comment]
no time to go through all the projects now, I will stop working on this for a while. I think it is worth the trouble though. Any additions/suggestions are welcome.

**Comment by rohityadavcloud** [comment]
@DaanHoogland if codecov, sonar are enough should we drop this? I had proposed this but recently I've found the project backers were let go and this may require more effort than initially thought.

**Comment by rohityadavcloud** [comment]
Thanks @DaanHoogland given we've sonar and codecov, we can revisit this in future; you may close the PR.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java [hunk 1]
```diff
@@ -87,15 +87,15 @@ public void validateAddressAndPrepareForUrlTestIpv6() {
     public void buildRequestUrlTestHttpsGetSystemId() {
         RedfishClient redfishclient = new RedfishClient(USERNAME, PASSWORD, true, false, REDFISHT_REQUEST_RETRIES);
         String result = redfishclient.buildRequestUrl(oobAddress, RedfishClient.RedfishCmdType.GetSystemId, systemId);
-        String expected = String.format("https://%s/redfish/v1/Systems/", oobAddress, systemId);
```
Comment (weizhouapache) [inline utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java:90]: should not `systemId` be a part of the string ?

Comment (DaanHoogland) [reply utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java:90]: hm, not intended change. I will investigate why this creeped in.

Comment (DaanHoogland) [reply utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java:90]: on second thought , I just assumed it worked with the resulting string so the parameter wasn't needed. No idea what redfish actually requires.

Comment (weizhouapache) [reply utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java:90]: I guess the pattern is `"https://%s/redfish/v1/Systems/%s"

File: utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java [hunk 2]
```diff
@@ -87,15 +87,15 @@ public void validateAddressAndPrepareForUrlTestIpv6() {
...
+        String expected = String.format("https://%s/redfish/v1/Systems/", oobAddress);
```
Comment (DaanHoogland) [inline utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java:90]:
```suggestion
        String expected = String.format("https://%s/redfish/v1/Systems/%s", oobAddress, systemId);
```

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java (hunk 1 above)
Comment (DaanHoogland) [reply utils/src/test/java/org/apache/cloudstack/utils/redfish/RedfishClientTest.java:90]: @weizhouapache the tests actually fail this way. Not sure if this is a bug in the test or in the redfish code (`redfishclient.buildRequestUrl(oobAddress, RedfishClient.RedfishCmdType.GetSystemId, systemId);`)

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by shwstppr — APPROVED** [review]
code change look okay

**Comment by DaanHoogland** [comment]
let's test again
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by vishesh92 — APPROVED** [review]
Code lgtm

**Pull request merged**